src/lib/pickles/fix_domains.ml

```diff
@@ -16,10 +16,11 @@ let rough_domains : Domains.t =
   let d = Domain.Pow_2_roots_of_unity 20 in
   { h = d; x = Pow_2_roots_of_unity 6 }
 
-let domains (type field)
+let domains (type field rust_gates)
     (module Impl : Snarky_backendless.Snark_intf.Run
       with type field = field
-       and type R1CS_constraint_system.t = field
+       and type R1CS_constraint_system.t = ( field
+                                           , rust_gates )
```

Comment:
unfortunately, because rust_gates is defined as an interface passed to the constraint system functor (and parameterized based on the field), we have to parameterize `t` with `rust_gates` as well.

If we got rid of that rust vector type (and just played with an OCaml list of gates), we could get rid of this extra parameter as well. If the reader of this comment thinks it's a good idea, we should create an issue for that.